src/pudl/settings.py

```diff
@@ -459,11 +473,99 @@ def validate_tables(cls, tables):  # noqa: N805
+
+    Args:
+        years: List of years to validate.
+    """
+
+    data_source: ClassVar[DataSource] = DataSource.from_id("ferc6")
+    years: list[int] = data_source.working_partitions["years"]
+    taxonomy: AnyHttpUrl = "https://eCollection.ferc.gov/taxonomy/form6/2022-01-01/form/form6/form-6_2022-01-01.xsd"
+
+
+class Ferc60XbrlToSqliteSettings(FercGenericXbrlToSqliteSettings):
+    """An immutable pydantic model to validate FERC from 60 XBRL to SQLite settings.
+
+    Args:
+        years: List of years to validate.
+    """
+
+    data_source: ClassVar[DataSource] = DataSource.from_id("ferc60")
+    years: list[int] = data_source.working_partitions["years"]
+    taxonomy: AnyHttpUrl = "https://eCollection.ferc.gov/taxonomy/form60/2022-01-01/form/form60/form-60_2022-01-01.xsd"
+
+
+class Ferc714XbrlToSqliteSettings(FercGenericXbrlToSqliteSettings):
+    """An immutable pydantic model to validate FERC from 714 XBRL to SQLite settings.
+
+    Args:
+        years: List of years to validate.
+    """
+
+    data_source: ClassVar[DataSource] = DataSource.from_id("ferc714")
+    years: list[int] = [2021]
+    taxonomy: AnyHttpUrl = "https://eCollection.ferc.gov/taxonomy/form714/2022-01-01/form/form714/form-714_2022-01-01.xsd"
+
+
+class FercToSqliteSettings(BaseSettings):
+    """An immutable pydantic model to validate FERC XBRL to SQLite settings.
+
+    Args:
+        ferc1_dbf_to_sqlite_settings: Settings for converting FERC 1 DBF data to SQLite.
+        ferc1_xbrl_to_sqlite_settings: Settings for converting FERC 1 XBRL data to SQLite.
+        other_xbrl_forms: List of non-FERC1 forms to convert from XBRL to SQLite.
+    """
 
     ferc1_dbf_to_sqlite_settings: Ferc1DbfToSqliteSettings = None
     ferc1_xbrl_to_sqlite_settings: Ferc1XbrlToSqliteSettings = None
+    ferc2_xbrl_to_sqlite_settings: Ferc2XbrlToSqliteSettings = None
+    ferc6_xbrl_to_sqlite_settings: Ferc6XbrlToSqliteSettings = None
+    ferc60_xbrl_to_sqlite_settings: Ferc60XbrlToSqliteSettings = None
+    ferc714_xbrl_to_sqlite_settings: Ferc714XbrlToSqliteSettings = None
+
+    def get_xbrl_dataset_settings(
+        self, form_number: int
+    ) -> FercGenericXbrlToSqliteSettings:
+        """Return a list with all requested FERC XBRL to SQLite datasets.
+
+        Args:
+            form_number: Get settings by FERC form number.
+        """
+        # Get proper settings object
+        if form_number == 1:
+            settings = self.ferc1_xbrl_to_sqlite_settings
+        elif form_number == 2:
+            settings = self.ferc2_xbrl_to_sqlite_settings
+        elif form_number == 6:
+            settings = self.ferc6_xbrl_to_sqlite_settings
+        elif form_number == 60:
+            settings = self.ferc60_xbrl_to_sqlite_settings
+        elif form_number == 714:
+            settings = self.ferc714_xbrl_to_sqlite_settings
+        else:
+            raise ValueError(f"{form_number} is not a valid FERC form number.")
```

Comment:
A pattern I like in lieu of this kind of if/else casing looks like

```py
def get_xbrl_dataset_settings(self, form_number: int) -> FercGenericXbrlToSqliteSettings:
    xbrl_settings: dict[str, FercGenericXbrlToSqliteSettings] = {
        1: self.ferc1_xbrl_to_sqlite_settings,
        2: self.ferc2_xbrl_to_sqlite_settings,
        6: self.ferc6_xbrl_to_sqlite_settings,
        60: self.ferc60_xbrl_to_sqlite_settings,
        714: self.ferc714_xbrl_to_sqlite_settings,
    }
    try:
        return xbrl_settings[form_number]
    except KeyError:
        logger.error(f"{form_number} is not a valid FERC form number.")
        raise
```

Reply:
Since we are only supporting 3.10 now, how do feel about using python's new [pattern matching](https://peps.python.org/pep-0636/)

Reply:
Seems great!